include/aspect/material_model/non_dimensional.h

```diff
@@ -0,0 +1,146 @@
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef _aspect_material_model_nondimensional_h
+#define _aspect_material_model_nondimensional_h
+
+#include <aspect/material_model/interface.h>
+#include <aspect/simulator_access.h>
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    using namespace dealii;
+
+    /**
+     * A material model for incompressible (using the boussinesq approximation)
+     * and compressible computations (with ALA or TALA) for a nondimensionalized
+     * problem. The viscosity is (optionally) depth and temperature dependent.
+     *
+     * @ingroup MaterialModels
+     */
+    template <int dim>
+    class Nondimensional : public MaterialModel::Interface<dim>, public ::aspect::SimulatorAccess<dim>
+    {
+      public:
+
+        /**
+         * Initialize.
+         */
+        virtual
+        void initialize ();
+
+        /**
+         * Evaluate the material model.
+         */
+        virtual void evaluate(const MaterialModel::MaterialModelInputs<dim> &in,
+                              MaterialModel::MaterialModelOutputs<dim> &out) const;
+
+        /**
+         * @name Qualitative properties one can ask a material model
+         * @{
+         */
+
+        /**
+         * Return whether the model is compressible or not.
+         */
+        virtual bool is_compressible () const;
+        /**
+         * @}
+         */
+
+        /**
+         * @name Reference quantities
+         * @{
+         */
+        virtual double reference_viscosity () const;
+
+        virtual double reference_density () const;
+
+        double reference_cp () const;
+        /**
+         * @}
+         */
+
+        /**
+         * @name Functions used in dealing with run-time parameters
+         * @{
+         */
+        /**
+         * Declare the parameters this class takes through input files.
+         */
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
+
+        /**
+         * Read the parameters this class declares from the parameter file.
+         */
+        virtual
+        void
+        parse_parameters (ParameterHandler &prm);
+        /**
+         * @}
+         */
+
+      private:
+        /**
+         * Use truncated anelastic approximation?
+         */
+        bool tala;
+
+        /**
+         * Are we ALA/TALA or BA? (determined by the choice of Di)
+         */
+        bool compressible;
+
+        /**
+         * Parameter describing the temperature prefactor of viscosity.
+         */
+        double exponential_viscosity_temperature_prefactor;
+        /**
+         * Parameter describing the depth prefactor of viscosity.
+         */
+        double exponential_viscosity_depth_prefactor;
+
+        /**
+         * The surface density.
+         */
+        double reference_rho;
+
+        /**
+         * The nondimensional numbers (Dissipation number,
+         * Rayleigh number, grueneisen parameter).
```

Comment:
I think if you spell "Rayleigh" with upper case, you ought to spell "Grueneisen" the same :-) Eduard Grüneisen would probably agree ;-)